File: v2/test/kitchensink/window.go
package main

import (
	wails "github.com/wailsapp/wails/v2"
)

// Window struct
type Window struct {
	runtime *wails.Runtime
}

// WailsInit is called at application startup
func (w *Window) WailsInit(runtime *wails.Runtime) error {
	// Perform your setup here
	w.runtime = runtime
	return nil
}

func (w *Window) SetTitle(title string) {
	w.runtime.Window.SetTitle(title)
}

func (w *Window) Fullscreen() {
	w.runtime.Window.Fullscreen()
}

func (w *Window) UnFullscreen() {
	w.runtime.Window.UnFullscreen()
}

func (w *Window) Maximise() {
	w.runtime.Window.Maximise()
}
func (w *Window) Unmaximise() {
	w.runtime.Window.Unmaximise()
}
func (w *Window) Minimise() {
	w.runtime.Window.Minimise()
}
func (w *Window) Unminimise() {
	w.runtime.Window.Unminimise()
}
func (w *Window) Center() {
	w.runtime.Window.Center()
}
func (w *Window) Show() {
	w.runtime.Window.Show()
}
func (w *Window) Hide() {
	w.runtime.Window.Hide()
}
func (w *Window) SetSize(width int, height int) {
	w.runtime.Window.SetSize(width, height)
}
func (w *Window) SetPosition(x int, y int) {
	w.runtime.Window.SetPosition(x, y)
}
func (w *Window) Close() {
	w.runtime.Window.Close()
}

File: v3/internal/api/src/window.ts
/*
 _	   __	  _ __
| |	 / /___ _(_) /____
| | /| / / __ `/ / / ___/
| |/ |/ / /_/ / / (__  )
|__/|__/\__,_/_/_/____/
The electron alternative for Go
(c) Lea Anthony 2019-present
*/

/* jshint esversion: 9 */

import {newRuntimeCaller} from "./runtime";

export function newWindow(id: number) {
    let call = newRuntimeCaller("window", id);
    return {
        // Reload: () => call('WR'),
        // ReloadApp: () => call('WR'),
        // SetSystemDefaultTheme: () => call('WASDT'),
        // SetLightTheme: () => call('WALT'),
        // SetDarkTheme: () => call('WADT'),
        // IsFullscreen: () => call('WIF'),
        // IsMaximized: () => call('WIM'),
        // IsMinimized: () => call('WIMN'),
        // IsWindowed: () => call('WIF'),
        Center: () => call('Center'),
        SetTitle: (title: string) => call('SetTitle', {title}),
        Fullscreen: () => call('Fullscreen'),
        UnFullscreen: () => call('UnFullscreen'),
        SetSize: (width: number, height:number) => call('SetSize', {width,height}),
        Size: () => { return call('Size') },
        SetMaxSize: (width:number, height:number) => call('SetMaxSize', {width,height}),
        SetMinSize: (width:number, height:number) => call('SetMinSize', {width,height}),
        SetAlwaysOnTop: (b:boolean) => call('SetAlwaysOnTop', {alwaysOnTop:b}),
        SetPosition: (x:number, y:number) => call('SetPosition', {x,y}),
        Position: () => { return call('Position') },
        Screen: () => { return call('Screen') },
        Hide: () => call('Hide'),
        Maximise: () => call('Maximise'),
        Show: () => call('Show'),
        Close: () => call('Close'),
        ToggleMaximise: () => call('ToggleMaximise'),
        UnMaximise: () => call('UnMaximise'),
        Minimise: () => call('Minimise'),
        UnMinimise: () => call('UnMinimise'),
        SetBackgroundColour: (r:number, g:number, b:number, a:number) => call('SetBackgroundColour', {r, g, b, a}),
    }
}
